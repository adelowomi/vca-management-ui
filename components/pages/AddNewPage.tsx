import { gql, useMutation } from '@apollo/client';
import Link from 'next/link';
import React from 'react';
import Select from 'react-select';
import makeAnimated from 'react-select/animated';

import DropdownRender from './Dropdown';

const animatedComponents = makeAnimated();
const tags = [
  { value: 'bank', label: 'banks' },
  { value: 'economy', label: 'economy' },
  { value: 'finance', label: 'finance' },
  { value: 'country', label: 'country' },
  { value: 'church', label: 'church' },
  { value: 'tech', label: 'tech' },
];

const ADD_PAGE = gql`
  mutation CreatePage($createPageInput: CreatePageInput!) {
    createPage(createPageInput: $createPageInput) {
      id
    }
  }
`;

const AddNewPage = ({ site }) => {
  console.log(site);
  const [header, setHeader] = React.useState('');
  const [state, setState] = React.useState({
    pageTitle: '',
    site,
    mediaUrl: '',
    headerText: '',
    captionText: '',
    actionText: '',
    bodyTitle: '',
    bodyDescription: '',
    ctaLink: '',
    headerType: '',
    tags: [],
    location: 'LEFT',
  });
  const [createPage, { data }] = useMutation(ADD_PAGE);

  const onButtonClick = (e: any) => {
    setHeader(e.target.value);
  };

  const handleChange = (e: any) => {
    const value = e.target.value;
    setState({
      ...state,
      [e.target.name]: value,
    });
  };

  const onSelect = (data: any) => {
    const arr = data.map((el) => el.value);
    setState({
      ...state,
      tags: arr,
    });
  };
  const onSubmit = (e: { preventDefault: () => void }) => {
    e.preventDefault();
    createPage({
      variables: {
        createPageInput: {
          name: state.pageTitle,
          tags: state.tags,
          site: state.site,
          hero: {
            caption: state.captionText,
            type: 'Bank',
            mediaUrl: state.mediaUrl,
            heading: state.headerText,
            hasAction: true,
            actoinText: state.actionText,
            actoinSlug: state.ctaLink,
            location: state.location,
          },
        },
      },
    });
  };

  React.useEffect(() => {
    setState({
      ...state,
      headerType: header,
    });
  }, [header]);
  // console.log(header);
  // console.log('STATE: ', state);

  return (
    <div className="px-5">
      <div className="breadCrumb">
        {/* <!-- This example requires Tailwind CSS v2.0+ --> */}
        <nav className="flex" aria-label="Breadcrumb">
          <ol className="flex items-center space-x-4">
            <li>
              <div className="flex items-center">
                {/* <!-- Heroicon name: solid/chevron-right --> */}
                <a
                  href="#"
                  className="font-medium text-base text-black hover:text-gray-700"
                >
                  Pages
                </a>
              </div>
            </li>
            <li>
              <div className="flex items-center">
                {/* <!-- Heroicon name: solid/chevron-right --> */}
                <svg
                  className="flex-shrink-0 h-5 w-5 text-gray-400"
                  xmlns="http://www.w3.org/2000/svg"
                  viewBox="0 0 20 20"
                  fill="currentColor"
                  aria-hidden="true"
                >
                  <path
                    fillRule="evenodd"
                    d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                    clipRule="evenodd"
                  />
                </svg>
                <Link href="#">
                  <a
                    aria-current="page"
                    className="ml-4 text-base font-medium text-gray-500 hover:text-gray-700"
                  >
                    Add New
                  </a>
                </Link>
              </div>
            </li>
          </ol>
        </nav>
      </div>
      <form action="" className="form mt-5">
        {/* <div className="grid grid-cols-3 gap-4"> */}
        <div className="flex flex-1 justify-between">
          <div className="w-72">
            <input
              type="text"
              name="pageTitle"
              value={state.pageTitle}
              onChange={handleChange}
              id="pageTitle"
              placeholder="Page Title"
              autoComplete="pageTitle"
              className="mt-1 block w-full border cursor-pointer border-gray-300 rounded-md shadow-sm py-3 px-3 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm placeholder-gray-600"
            />
          </div>
          <div className="">
            <div className="flex flex-1 items-center ">
              <div className="">
                <a
                  href="#"
                  className="text-base font-medium text-red-600 underline leading-6"
                >
                  Cancel all changes
                </a>
              </div>
              <div className="mx-10">
                <DropdownRender />
              </div>
              <div>
                <button className="px-5 py-3 text-lg font-medium rounded text-indigo-700 bg-indigo-100 leading-6">
                  Preview
                </button>
              </div>
            </div>
          </div>
        </div>
        <div className="headerSection">
          <div className="mt-8 mb-1">
            <h3 className="ml-3 text-sm ">Header</h3>
          </div>
          <div className="rounded-lg text-sm  bg-white overflow-hidden shadow  px-3">
            <div className="mt-4 mb-3">
              <h3 className="text-sm ">Style</h3>
            </div>
            <div className="header-button-group flex justify-start space-x-3 ">
              <button
                onClick={onButtonClick}
                value="headerTypeOne"
                type="button"
                className="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              >
                Header Type 1
              </button>
              <button
                value="headerTypeTwo"
                onClick={onButtonClick}
                type="button"
                className="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              >
                Header Type 2
              </button>
              <button
                value="headerTypeThree"
                onClick={onButtonClick}
                type="button"
                className="inline-flex items-center px-3 py-2 border  shadow-sm text-base font-medium rounded-md text-indigo-700 bg-indigo-200 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              >
                Header Type 3
              </button>
              <button
                value="headerTypeFour"
                onClick={onButtonClick}
                type="button"
                className="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
              >
                Header Type 4
              </button>
            </div>
            <div className="inputSection mt-6 grid grid-cols-7">
              <div className=" col-span-3">
                <label className="text-gray-700 font-medium">Media URL</label>
                <input
                  name="mediaUrl"
                  value={state.mediaUrl}
                  onChange={handleChange}
                  type="text"
                  placeholder="https://star-studded-bricks-1440by551.png"
                  className="w-full mt-2 mb-6 px-4 py-2 border rounded-lg text-gray-700 focus:outline-none focus:border-indigo-500"
                />
              </div>
            </div>
            <div className="inputSection2 mt-1 grid grid-cols-7">
              <div className=" col-span-3">
                <label className="text-gray-700 font-medium">Header Text</label>
                <input
                  type="text"
                  name="headerText"
                  value={state.headerText}
                  onChange={handleChange}
                  placeholder="RoloBank"
                  className="w-full mt-2 mb-6 px-4 py-2 border rounded-lg text-gray-700 focus:outline-none focus:border-indigo-500"
                />
              </div>
            </div>
            <div className="inputSection3 mt-1 ">
              <div className="">
                <label className="text-gray-700 font-medium">
                  Text Position
                </label>
                <div className="flex space-x-3 mt-2 text-gray-500">
                  <a
                    href="#"
                    className="focus:border-b-2 focus:border-blue-800 focus:text-blue-800  text-sm cursor-pointer "
                  >
                    Left
                  </a>
                  {/* <a
                    href="#"
                    className="text-sm text-blue-800 border-b-2 border-blue-800 focus:text-blue-800 cursor-pointer "
                  >
                    Right
                  </a> */}
                  <a
                    href="#"
                    className="text-sm focus:text-blue-800 cursor-pointer "
                  >
                    Right
                  </a>
                  <a
                    href="#"
                    className="text-sm cursor-pointer focus:text-blue-800"
                  >
                    Bottom
                  </a>
                </div>
              </div>
            </div>
            <div className="inputSection2 mt-5 grid grid-cols-7">
              <div className=" col-span-3">
                <label className="text-gray-700 font-medium">
                  Caption Text
                </label>
                <input
                  name="captionText"
                  value={state.captionText}
                  onChange={handleChange}
                  type="text"
                  placeholder="WE ARE A BANK OF THE YOUNG AND FOR YOUNG"
                  className="w-full mt-2 mb-6 px-4 py-2 border rounded-lg text-gray-700 focus:outline-none focus:border-indigo-500"
                />
              </div>
            </div>
            <div className="mt-1 mb-5 w-96">
              <label className="text-gray-700 font-medium">Tags</label>
              <Select
                // defaultValue={[tags[2], tags[3]]}
                isMulti
                name="tags"
                options={tags}
                className="basic-multi-select"
                classNamePrefix="select"
                components={animatedComponents}
                onChange={onSelect}
              />
            </div>
            <div className="inputSection2 mt-1 grid grid-cols-7">
              <div className=" col-span-3">
                <label className="text-gray-700 font-medium">
                  Call To Actions Button
                </label>
                <div className="flex space-x-2 mt-2">
                  <button
                    type="button"
                    className="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-400 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
                  >
                    Inactive
                  </button>
                  <input
                    style={{ background: '#F2F2F2', color: '#A3A3A3' }}
                    name="actionText"
                    value={state.actionText}
                    onChange={handleChange}
                    type="text"
                    placeholder="Action Text"
                    className="inline-flex font-light items-center  pl-3  border border-gray-300 shadow-sm text-sm italic rounded-md text-gray-500  focus:outline-none focus:border-indigo-500"
                  />
                </div>
                <input
                  style={{ background: '#F2F2F2', color: '#A3A3A3' }}
                  name="ctaLink"
                  value={state.ctaLink}
                  onChange={handleChange}
                  type="text"
                  placeholder="CTA (call to action)link"
                  className="w-full mt-2 mb-6 px-4 py-2 border rounded-lg italic text-gray-700 focus:outline-none focus:border-indigo-500"
                />
              </div>
            </div>
          </div>
          <button
            onClick={onSubmit}
            type="button"
            className="inline-flex mt-4 items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-white bg-green-400 hover:bg-green-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
          >
            persist all
          </button>
          {/* First preview section */}
          <div>
            <div className="mb-2">
              <h3 className="ml-3 text-sm ">Preview</h3>
            </div>
            <div className="rounded-lg text-sm  bg-white overflow-hidden shadow  px-3 h-40"></div>
          </div>
          {/* End of first Preview Section */}
          <div className="bodySection mt-12">
            <div className="mb-1">
              <h3 className="ml-3 text-sm ">Body</h3>
            </div>
            <div className="rounded-lg text-sm  bg-white overflow-hidden shadow  px-3">
              <div className="buttons space-x-3 mt-5 flex">
                <button
                  type="button"
                  className="inline-flex items-center px-6 py-2 border  shadow-sm text-base font-medium leading-7 rounded-md text-white bg-indigo-500  focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 focus:bg-white focus:text-gray-500"
                >
                  Add Widget
                </button>
                <button
                  type="button"
                  className="inline-flex items-center px-5 py-2 border  shadow-sm text-base font-medium rounded-md text-gray-500 bg-gray-100 focus:bg-indigo-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 focus:text-white"
                >
                  Add Item
                </button>
              </div>
              <div className="inputSection mt-6 grid grid-cols-7">
                <div className=" col-span-3">
                  <label className="text-gray-700 font-medium text-sm">
                    Title
                  </label>
                  <input
                    type="text"
                    name="bodyTitle"
                    value={state.bodyTitle}
                    onChange={handleChange}
                    className="w-full mt-2 px-4 py-2 border rounded-lg text-gray-700 focus:outline-none focus:border-green-500"
                  />
                </div>
              </div>
              <div className="inputSection mt-1 grid grid-cols-7">
                <div className=" col-span-3">
                  <label className="text-gray-700 font-medium text-sm">
                    Description
                  </label>
                  <input
                    name="bodyDescription"
                    value={state.bodyDescription}
                    onChange={handleChange}
                    type="text"
                    className="w-full mt-2  px-4 py-2 border rounded-lg text-gray-700 focus:outline-none focus:border-green-500"
                  />
                </div>
              </div>
              <div className="inputSection mt-1 mb-10 grid grid-cols-7">
                <div className=" col-span-3">
                  <label className="text-gray-700 font-medium text-sm">
                    Add Items
                  </label>
                  <button
                    type="button"
                    className="w-full mt-2 px-4 py-3 border bg-indigo-600 rounded-lg text-white  focus:outline-none focus:border-green-500"
                  >
                    Select from a list of posts
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className="mt-10">
            <div className="mb-2">
              <h3 className="ml-3 text-sm ">Preview</h3>
            </div>
            <div className="rounded-lg text-sm  bg-white overflow-hidden shadow  px-3 h-40"></div>
          </div>
        </div>
      </form>
    </div>
  );
};

export default AddNewPage;
