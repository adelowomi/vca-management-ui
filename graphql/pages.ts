import { gql } from '@apollo/client';

const PAGES_QUERY = gql`
  query Pages($limit: Int, $offset: Int, $filter: FilterInput,$accountId: String!) {
    pages(filter: $filter, limit: $limit, offset: $offset,accountId:$accountId) {
      id
      name
      tags
      site
      menuItem
      hero {
        type
        caption
        mediaUrl
        heading
        hasAction
        actionText
        actionSlug
        location
        media {
          image {
            small
            medium
            large
          }
        }
      }
    }
  }
`;
const DELETE_PAGE = gql`
  mutation ($pageId: String!) {
    removePage(pageId: $pageId) {
      id
    }
  }
`;
const PAGE_QUERY = gql`
  query Page($accountId:String!,$filter: FilterInput) {
    page(filter: $filter,accountId: $accountId) {
      id
      name
      tags
      site
      menuItem
      createdAt
      hero {
        media {
          image {
            small
            medium
            large
          }
        }
        type
        caption
        mediaUrl
        heading
        hasAction
        actionText
        actionSlug
        location
      }
    }
  }
`;

const ADD_PAGE = gql`
  mutation CreatePage($createPageInput: CreatePageInput!) {
    createPage(createPageInput: $createPageInput) {
      id
      name
      tags
      hero {
        type
        caption
        mediaUrl
        heading
        hasAction
        actionText
        actionSlug
        location
      }
    }
  }
`;
const EDIT_PAGE = gql`
  mutation ($updatePageInput: UpdatePageInput!, $pageId: String!) {
    updatePage(updatePageInput: $updatePageInput, pageId: $pageId) {
      id
      name
      tags
      site
      menuItem
      hero {
        type
        caption
        mediaUrl
        heading
        hasAction
        actionText
        actionSlug
        location
      }
    }
  }
`;
export { ADD_PAGE, DELETE_PAGE, EDIT_PAGE, PAGE_QUERY, PAGES_QUERY };
