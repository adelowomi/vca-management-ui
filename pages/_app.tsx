import '../styles/global.css';

import { gql } from '@apollo/client';
import { getSession, UserProvider } from '@auth0/nextjs-auth0';
import { BaseProvider } from 'baseui';
import App, { AppProps } from 'next/app';
import Head from 'next/head';
import { useRouter } from 'next/router';
import React from 'react';
import { IntlProvider } from 'react-intl';
import styled, { ThemeProvider } from 'styled-components';
import { Provider as StyletronProvider } from 'styletron-react';

import * as locales from '../content/locale';
import { AppWrapper } from '../context/state';
import { ApolloProvider, createApolloClient, useApollo } from '../lib/apollo';
import { GlobalStyles, theme } from '../styles';
import { appThemes } from '../styles/baseui-theme';
import { styletron } from '../styletron';

const Container = styled.div`
  margin: 0 auto;
  font-family: 'Avenir Next';
`;

const SITE_QUERY = gql`
  query Sites {
    sites {
      id
      name
      header {
        name
        type
        menuItems {
          id
          name
          slug
          description
        }
      }
    }
  }
`;

const MyApp = ({ Component, pageProps }: AppProps) => {
  const { locale, defaultLocale } = useRouter();
  const messages = locales[locale];
  const apolloClient = useApollo(
    pageProps.initialApolloState,
    pageProps.idToken
  );

  // it is possible to have errors when the site is retiieved so we
  // handle the error TODO: create an error page for this error
  if (pageProps.sites?.error || pageProps.sites?.errors) {
    return (
      <UserProvider>
        <ApolloProvider client={apolloClient}>
          <StyletronProvider value={styletron}>
            <BaseProvider theme={appThemes}>
              <ThemeProvider theme={theme()}>
                <IntlProvider
                  locale={locale}
                  defaultLocale={defaultLocale}
                  messages={messages}
                >
                  <Head>
                    <title>Qaltrak</title>
                    <meta name="Description" content="Qaltrak" />
                  </Head>
                  <Container>
                    <GlobalStyles />
                    <div>{`${pageProps.sites?.error}`}</div>
                    <div>{`${pageProps.sites?.errors}`}</div>
                  </Container>
                </IntlProvider>
              </ThemeProvider>
            </BaseProvider>
          </StyletronProvider>
        </ApolloProvider>
      </UserProvider>
    );
  }

  return (
    <AppWrapper value={pageProps.sites.data['sites'][0]}>
      <UserProvider>
        <ApolloProvider client={apolloClient}>
          <StyletronProvider value={styletron}>
            <BaseProvider theme={appThemes}>
              <ThemeProvider theme={theme()}>
                <IntlProvider
                  locale={locale}
                  defaultLocale={defaultLocale}
                  messages={messages}
                >
                  <Head>
                    <title>Qaltrak</title>
                    <meta name="Description" content="Qaltrak" />
                  </Head>
                  <Container>
                    <GlobalStyles />
                    <Component {...pageProps} />
                  </Container>
                </IntlProvider>
              </ThemeProvider>
            </BaseProvider>
          </StyletronProvider>
        </ApolloProvider>
      </UserProvider>
    </AppWrapper>
  );
};

MyApp.getInitialProps = async (appContext) => {
  const session = await getSession(appContext.ctx.req, appContext.ctx.res);
  const appProps = await App.getInitialProps(appContext);

  if (!session) {
    return { pageProps: {} };
  }
  const client = createApolloClient(session.idToken);

  // Each user should have only one site and we need to retrieve it in order
  // to know what site we are configuring in the management UI
  const sites = await client.query({ query: SITE_QUERY });
  return {
    pageProps: {
      ...appProps.pageProps,
      ...session,
      sites,
    },
  };
};

export default MyApp;
