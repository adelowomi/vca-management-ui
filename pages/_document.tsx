import Document, { DocumentContext } from 'next/document';
import { ServerStyleSheet } from 'styled-components';
import { Provider as StyletronProvider } from 'styletron-react';

import { styletron } from '../styletron';

export default class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const sheet = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            sheet.collectStyles(
              <StyletronProvider value={styletron}>
                <App {...props} />
              </StyletronProvider>
            ),
        });

      const initialProps = await Document.getInitialProps(ctx);

      // @ts-expect-error for styletron
      const stylesheets = styletron.getStylesheets() || [];
      return {
        ...initialProps,
        stylesheets,
        styles: (
          <>
            {initialProps.styles}
            {sheet.getStyleElement()}
          </>
        ),
      };
    } finally {
      sheet.seal();
    }
  }
}
