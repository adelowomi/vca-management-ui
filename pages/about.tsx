import React from 'react';

const about: React.FC = () => {
  return (
    <div>
      <h1>Welcome to about Us</h1>
    </div>
  );
};

export default about;
