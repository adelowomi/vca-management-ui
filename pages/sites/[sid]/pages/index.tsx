const index = () => <div>List of pages</div>;
export default index;
