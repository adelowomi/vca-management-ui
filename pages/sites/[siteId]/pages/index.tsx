import { getSession } from '@auth0/nextjs-auth0';
import { withPageAuthRequired } from '@auth0/nextjs-auth0/dist/frontend';
// import moment from 'moment';
// import Link from 'next/link';
import React from 'react';
import styled from 'styled-components';
import tw from 'tailwind-styled-components';

import Layout from '../../../../components/Layout/Layout';
import { BulkActionDropdown } from '../../../../components/Page/Create/BulkActionDropdown';
// import { PAGES_QUERY } from '../../../../graphql';
// import { createApolloClient } from '../../../../lib/apollo';
interface Iprops {
  $big: boolean;
}

const PageActionsWrapper = tw.div`
flex
flex-col
`;

const PageActionsColOne = tw.div`
flex
flex-row
justify-between
items-center
`;

const Btn = styled.button`
  background-color: #1890ff;
  border-radius: 4px;
  box-shadow: rgba(128, 128, 128, 0.1);
`;

const PageActionsColOneBtn = tw(Btn)`
text-white
text-sm
py-3
font-bold
px-7
`;

const PageHeroWrapper = tw.div``;
const PageTableWrapper = tw.div``;

const Pages = () => {
  // const count = 0;
  // const [dropdown, setDropdown] = React.useState(false);
  // const dropdownClick = () => setDropdown(!dropdown);
  return (
    <Layout>
      <PageActionsWrapper>
        <PageActionsColOne>
          <h1 className="text-4xl font-semibold">Pages</h1>
          <PageActionsColOneBtn>Add New</PageActionsColOneBtn>
        </PageActionsColOne>
        <div className=" flex mt-7">
          <BulkActionDropdown />
        </div>
      </PageActionsWrapper>
      <PageHeroWrapper></PageHeroWrapper>
      <PageTableWrapper></PageTableWrapper>
    </Layout>
  );
};

export async function getServerSideProps(ctx) {
  const session = getSession(ctx.req, ctx.res);

  if (!session) {
    return {
      redirect: {
        destination: '/login',
      },
    };
  }
  // const client = createApolloClient(session.idToken);

  // const {
  //   data: { pages },
  // } = await client.query({
  //   query: PAGES_QUERY,
  //   variables: {
  //     filter: {
  //       combinedFilter: {
  //         filters: [
  //           {
  //             singleFilter: {
  //               field: 'siteId',
  //               operator: 'EQ',
  //               value: ctx.query.siteId,
  //             },
  //           },
  //         ],
  //       },
  //     },
  //   },
  // });

  // return { props: { pages } };
  return { props: {} };
}

export default withPageAuthRequired(Pages);
