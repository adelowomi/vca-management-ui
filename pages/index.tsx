import { getSession, Session, withPageAuthRequired } from '@auth0/nextjs-auth0';
import { GetServerSideProps } from 'next';
import { useRouter } from 'next/router';
import * as React from 'react';
import styled from 'styled-components';

import { Button, Header } from '../components';
import { GqlErrorResponse } from '../errors/GqlError';
import { SITES_QUERY } from '../graphql';
import { createApolloClient } from '../lib/apollo';

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  flex-direction: column;
`;

export const Home = ({
  user,
  sites,
}: {
  user: any;
  sites: any;
}): JSX.Element => {
  const router = useRouter();
  // eslint-disable-next-line no-console
  console.log(sites);

  if (user) {
    return (
      <Container>
        <Header user={user} />
        {user && <Button width={'345px'}>Signed in as {user.name}</Button>}
        <div>
          <h1>Index page</h1>
          <p>Current locale: {router?.locale}</p>
          <p>Default locale: {router?.defaultLocale}</p>
          <p>Configured locales: {JSON.stringify(router?.locales)}</p>
        </div>
      </Container>
    );
  }
  return <Container>Unauthenticated</Container>;
};
export default withPageAuthRequired(Home);

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const session: Session = getSession(ctx.req, ctx.res);

  if (!session) {
    ctx.res.writeHead(302, {
      Location: '/login',
    });
    ctx.res.end();
    return;
  }

  const token = session.idToken;
  const client = createApolloClient(token);

  try {
    const sites = await client.query({
      query: SITES_QUERY,
      variables: {
        accountId: process.env.ACCOUNT_ID,
      },
    });
    return {
      props: {
        sites: sites.data.sites,
        error: null,
        user: session.user,
      },
    };
  } catch (error) {
    console.error(error);
    return {
      props: {
        sites: [],
        error: GqlErrorResponse(error),
        user: session.user,
      },
    };
  }
};
