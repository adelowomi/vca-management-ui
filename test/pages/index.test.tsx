import React from 'react';

import { Home } from '../../pages/index';
import { render } from '../testUtils';

describe('Home page', () => {
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <Home
        user={{
          nickname: 'nickname',
          name: 'name',
          picture: 'picture',
          sub: 'sub',
        }}
      />,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
