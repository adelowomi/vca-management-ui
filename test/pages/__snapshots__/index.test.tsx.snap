// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  height: 50px;
  width: 100px;
  margin: 8px 0px;
  border: none;
  border-radius: 5px;
  font-size: 14px;
  font-weight: 600;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  -webkit-letter-spacing: 0.71px;
  -moz-letter-spacing: 0.71px;
  -ms-letter-spacing: 0.71px;
  letter-spacing: 0.71px;
  text-align: start;
  background: none;
  color: #0035c1;
  cursor: pointer;
}

.c2:disabled {
  background: lightgray;
}

.c4 {
  height: 50px;
  width: 345px;
  margin: 8px 0px;
  border: none;
  border-radius: 5px;
  font-size: 17px;
  font-weight: 600;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  -webkit-letter-spacing: 0.71px;
  -moz-letter-spacing: 0.71px;
  -ms-letter-spacing: 0.71px;
  letter-spacing: 0.71px;
  text-align: center;
  background: #0035c1;
  color: #fff;
  cursor: pointer;
}

.c4:disabled {
  background: lightgray;
}

.c1 {
  height: 55px;
  padding: 0 16px;
  box-shadow: 0 1px 4px 0 #dfe5e9;
  background-color: #ffffff;
  position: absolute;
  width: 100%;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  padding: 8px;
  height: unset;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="logo"
        height="35"
        loading="lazy"
        src="/images/logo-colored-on-white.svg"
        width="126"
      />
      <a
        href="/api/logout"
      >
        <button
          aria-label="Logout"
          class="c2 c3"
          width="100px"
        >
          Logout
        </button>
      </a>
    </div>
    <button
      class="c4"
      width="345px"
    >
      Signed in as name
    </button>
    <div>
      <h1>
        Index page
      </h1>
      <p>
        Current locale: 
      </p>
      <p>
        Default locale: 
      </p>
      <p>
        Configured locales: 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
